### PR TITLE
Add OPA post

### PR DESCRIPTION
**Pull request by sbueringer**

### CODE REVIEW

File: _posts/2019-01-04-open-policy-agent-authorization-webhook.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+Notes:
+* We have to exclude `system:kube-controller-manager` and `system:kube-scheduler`, because both the Kubernetes controller manager as well as the scheduler have to be able to access pods.
+* By simply removing `resource.spec.resourceAttributes.resource = "pods"` from the policy, we could restrict access to all namespaced resources in `kube-system`. 
+* We have to be careful to deny the correct verbs. A simple `delete` in RBAC allows `delete` and  `deletecollection` (cf. [Authorization Overview](https://kubernetes.io/docs/reference/access-authn-authz/authorization/)).
+* Open Policy Agent makes it also very easy to write unit tests for all our policies. For more information, see [How Do I Test Policies?](https://www.openpolicyagent.org/docs/how-do-i-test-policies.html)
+
+## Create/ update/ delete on specific StorageClasses 
+
+As in the first example, we first have to grant user access via RBAC:
```
Comment (chrischdi) [inline _posts/2019-01-04-open-policy-agent-authorization-webhook.md]: "first...first" - I would remove the second one :-)

File: _posts/2019-01-04-open-policy-agent-authorization-webhook.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+# Conclusion
+
+In summary, OPA allows far more flexible policies compared to the built-in RBAC authorization, especially if no additional operator is used. In my opinion it would be nice to have direct integration for OPA as Authorization Module and Admission Controller, but in the meantime Kubernetes Policy Controller bridges the gap between Kubernetes and OPA. Some inspirations what can be implemented:
+* Deny access to specific CustomResourceDefinitions, e.g. `calico`
+* Deny access to specific ClusterRoles, e.g. `cluster-admin`, `admin`, `edit`, `view`
+* Allow port-forward only to some specific pods in `kube-system`
+* Create a mapping which PodSecurityPolicies can be used in which namespaces
+* Allow access to ValidatingWebhookConfigurations except some pre-installed ones
+
+What do you think about Open Policy Agent as policy engine for Kubernetes? What are your use-cases and are they already covered by RBAC? If not, what would you like to implement via the Open Policy Agent?
```
Comment (chrischdi) [inline _posts/2019-01-04-open-policy-agent-authorization-webhook.md:200]: I would add a reference to get in contact via Twitter or slack or something else

File: _posts/2019-01-04-open-policy-agent-authorization-webhook.md (hunk 1 above)
Comment (sbueringer) [reply _posts/2019-01-04-open-policy-agent-authorization-webhook.md]: fixed

File: _posts/2019-01-04-open-policy-agent-authorization-webhook.md (hunk 2 above)
Comment (sbueringer) [reply _posts/2019-01-04-open-policy-agent-authorization-webhook.md:200]: fixed

**Pull request merged**